### ISSUE TITLE
JSObjectPropertiesMap - proper usage 

### ISSUE THREAD

**Issue by bhickey777** [LiquidPlayer/LiquidCore#17]
It looks like this is an inner wrapper class to assist with maps. If I set a map in context as follows: 

        Map<String, Object> pageContext =  new HashMap<String, Object>();
        JSObject oPageContext = new JSObject(this.jsContext, pageContext);
        jsContext.property("pageContext", oPageContext);

Is this the proper way to then reference that map in the context (after running evaluate script and associated call function):

        JSValue _pageContext = jsContext.property("pageContext");
        Map<String, Object> map = (Map<String, Object> _pageContext.toObject();

thx, appreciate the assist.

**Comment by ericwlange** [comment]
It seems that I have mistakenly made `JSObjectPropertiesMap` package local.  It _should_ be `public`.  I will fix that for the next release.  Unfortunately, it will be difficult to use without reflections unless it is public.

The intent was to enable a convenient method of setting `JSObject` properties using the `Map` interface.  So the correct way to use it **_should_** be:

```java
JSValue _pageContext = jsContext.property("pageContext");
Map<String, JSValue> map = new JSObjectPropertiesMap<>(_pageContext.toObject(), JSValue.class);
```

I recommend using `JSValue` as a generic type rather than `Object` because not all objects can be translated into Javascript values.  Or, if your object uses only one type, you can specify that type, as in: `Map<String, Integer> map = new JSObjectPropertiessMap<>(_pageContext, Integer.class);`

But because the class is not public, this won't work.  If you really want to use it, there are two choices:  (1) add `public` in front of the class in `JSObjectPropertiesMap.java` and build the library yourself, or (2) use reflections:

```java
Class<?> cl = Class.forName("org.liquidplayer.javascript.JSObjectPropertiesMap");
Constructor<?> cons = cl.getConstructor(JSObject.class, Class.class);
Object inst = cons.newInstance(_pageContext.toObject(), JSValue.class);
Map<String, JSValue> map = (Map<String, JSValue>) inst;
```
You'll also need to wrap this in a `try/catch` block because there are several exceptions that can theoretically occur, although if your parameters are right, it should work fine.

As an aside, I had intended to just allow `JSObject` to always implement the `Map` inteface, however, in Java, I learned, an object cannot implement _both_ `Map` and `List`!!  This is because they share a single method, `remove`, which has different return types in each implementation!  I wanted `JSArray` (which is also a `JSObject`) be able to be used as a generic `List`.  And so I couldn't do both.  So to still make this possible, `JSObjectPropertiesMap` is just a shadow object for some `JSObject`.  Just a little trivia for you.

I will mark this as a bug and fix.  Thanks for reporting.

**Comment by ericwlange** [comment]
Fixed in [8cbe527e66dd7c57acc23f02c44be3d7f9bbdaab].(https://github.com/LiquidPlayer/LiquidCore/commit/8cbe527e66dd7c57acc23f02c44be3d7f9bbdaab)

**Comment by bhickey777** [comment]
Sounds good, for now I will mark class public and run a build. Appreciate it.

**Comment by ericwlange** [comment]
This is fixed in v0.2.1.
